python/lsst/rubintv/production/allSky.py

```diff
@@ -104,7 +108,70 @@ def dayObsFromDirName(fullDirName, logger):
         return None, None
 
 
-def _convertJpgScale(inFilename, outFilename):
+def getUbuntuFontPath(logger=None):
+    """Get the path to the Ubuntu font, if available.
+
+    Parameters
+    ----------
+    logger : `logging.logger`
+        The logger.
+
+    Returns
+    -------
+    ubuntuBoldPath : `str`
+        The path to the Ubuntu bold font, or ``None`` if not found.
+    """
+    fontPaths = matplotlib.font_manager.findSystemFonts(fontpaths=None, fontext='ttf')
+
+    ubuntuBoldPath = [f for f in fontPaths if f.find('Ubuntu-B.') != -1]
+    if not ubuntuBoldPath:
+        if not logger:  # only create if needed
+            logger = _LOG.getChild("getUbuntuFontPath")
+        logger.warning('Warning - cound not fund Ubuntu bold font!')
+        return
+    if len(ubuntuBoldPath) != 1:
+        if not logger:  # only create if needed
+            logger = _LOG.getChild("getUbuntuFontPath")
+        logger.warning('Warning - found multiple fonts for Ubuntu bold, picking the first!')
+    ubuntuBoldPath = ubuntuBoldPath[0]
+    return ubuntuBoldPath
+
+
+def getDateTimeFromExif(filename, logger=None):
+    """Get the image date and time from the exif data.
+
+    Parameters
+    ----------
+    filename : `str`
+        The filename to get the exif data from.
+
+    Returns
+    -------
+    dateStr, timeStr : `tuple` of `str`
+        The date and time strings, or ``None`` if parsing failed.
+    logger : `logging.logger`
+        The logger, created if needed and not supplied.
+    """
+    tagMap = {v: k for k, v in TAGS.items()}
+
+    with Image.open(filename) as img:
+        exifData = img.getexif()
+        tagNum = tagMap['DateTime']
+        dateTimeStr = exifData.get(tagNum)
+        if dateTimeStr:
+            # dateTimeStr comes out like "2021:08:17 06:57:00"
+            dateStr, timeStr = dateTimeStr.split(' ')
+            year, month, day = dateStr.split(':')
+            dateStr = f"{year}-{month}-{day}"
+            return dateStr, timeStr
+        else:
+            if not logger:  # only create if needed
+                logger = _LOG.getChild("getDateTimeFromExif")
+                logger.warning(f"Failed to get DateTime from exif data in {filename}")
```

Comment:
I think you want this line outside of the `if` block.

Reply:
Good point, thanks!